### PR TITLE
Add Remaining Parts of Service Functionality

### PR DESCRIPTION
**Pull request by farshad68**
- Add postman collection
- Stop Container
- Finish Experiment

### CODE REVIEW

File: src/main/java/eu/enexa/docker/ContainerManagerImpl.java [hunk 1]
```diff
@@ -156,10 +154,27 @@ public String startContainer(String image, String containerName, List<AbstractMa
             // add extra requirment based on the image name
             dockerHostConfig = addExceptionalConditions(image, allBinds, dockerHostConfig);
 
+            Map<String, String> labels = new HashMap<>();
+            labels.put("traefik.enable","true");
+            labels.put("traefik.http.routers."+containerName+".rule","Host(\"enexa-demo.cs.uni-paderborn.de\") && PathPrefix(\"/"+containerName+"\")");
```
Comment (MichaelRoeder) [inline src/main/java/eu/enexa/docker/ContainerManagerImpl.java]: `enexa-demo.cs.uni-paderborn.de` should be configurable, i.e., come from outside of this class.

The whole set of traefik labels should only be added, if the host URL of traefik is defined.

File: src/main/java/eu/enexa/docker/ContainerManagerImpl.java [hunk 2]
```diff
@@ -156,10 +154,27 @@ public String startContainer(String image, String containerName, List<AbstractMa
...
+
+            labels.put("traefik.http.middlewares."+containerName+"-prefix.stripprefix.prefixes","/"+containerName);
+            labels.put("traefik.http.middlewares."+containerName+"-prefix.stripprefix.forceSlash","true");
+
+            labels.put("traefik.http.routers."+containerName+".middlewares",containerName+"-prefix");
+
+
+//            LOGGER.info("traefik.http.routers."+containerName+".entrypoints:web");
+            labels.put("traefik.http.services."+containerName+".loadbalancer.server.port","8501");
+            LOGGER.info("traefik.http.services."+containerName+".loadbalancer.server.port:8501");
```
Comment (MichaelRoeder) [inline src/main/java/eu/enexa/docker/ContainerManagerImpl.java]: `8501` should be configurable.

File: src/main/java/eu/enexa/module/FileBasedModuleManager.java [hunk 3]
```diff
@@ -46,6 +46,12 @@ public FileBasedModuleManager() throws IOException {
                 LOGGER.info(enexaModuleDirectory.getAbsolutePath() + " exist");
                 addFileOrDirectory(enexaModuleDirectory);
                 LOGGER.info("modules size is : "+ modules.size());
+                for (Map.Entry<String, ModuleModel> entry : modules.entrySet()) {
+                    String key = entry.getKey();
+                    ModuleModel module = entry.getValue();
+                    System.out.println("Key: " + key + ", Value: " + module.toString());
+                }
+
```
Comment (MichaelRoeder) [inline src/main/java/eu/enexa/module/FileBasedModuleManager.java]: If this information is needed for debugging, it should be logged using a Logger.

If it was there only temporarily, it should be removed.

File: src/main/java/eu/enexa/service/EnexaServiceImpl.java [hunk 4]
```diff
@@ -242,22 +243,34 @@ public Model containerStatus(String experimentIri, String instanceIRI) {
     }
 
     @Override
-    public Model stopContainer(String experimentIri, String containerIri) {
+    public Model stopContainer(String experimentIri, String containerID) {
         Model model = ModelFactory.createDefaultModel();
+        String ResultOfDtoppingTheContainer = containerManager.stopContainer(containerID);
```
Comment (MichaelRoeder) [inline src/main/java/eu/enexa/service/EnexaServiceImpl.java]: 1. the variable should start lowercased
2. It should be "stopping"

File: src/main/java/eu/enexa/service/EnexaServiceImpl.java [hunk 5]
```diff
@@ -242,22 +243,34 @@ public Model containerStatus(String experimentIri, String instanceIRI) {
...
+        Model result = ModelFactory.createDefaultModel();
+        // TODO : check this part do we need an IRI or ID ?
+        Resource instance = result.createResource(containerID);
+        result.add(instance, ENEXA.experiment, result.createResource(experimentIri));
+        return result;
+    }
+
+    @Override
+    public Model finishExperiment(String experimentIri) {
+        // TODO Auto-generated method stub
```
Comment (MichaelRoeder) [inline src/main/java/eu/enexa/service/EnexaServiceImpl.java]: Please remove this TODO (and other TODOs that are fulfilled)

File: src/main/java/eu/enexa/sparql/SparqlBasedMetadataManager.java [hunk 6]
```diff
@@ -134,4 +136,20 @@ public String getContainerName(String experimentIri, String instanceIRI) {
         return null;
     }
 
+    @Override
+    public List<String> getAllContainersName(String experimentIri) {
```
Comment (MichaelRoeder) [inline src/main/java/eu/enexa/sparql/SparqlBasedMetadataManager.java]: Shouldn't it be called `getAllContainerNames`?

**Review by MichaelRoeder — CHANGES_REQUESTED** [review]
The code looks already good. However, some smaller changes are necessary.

File: src/main/java/eu/enexa/docker/ContainerManagerImpl.java (hunk 1 above)
Comment (farshad68) [reply src/main/java/eu/enexa/docker/ContainerManagerImpl.java]: done

File: src/main/java/eu/enexa/docker/ContainerManagerImpl.java (hunk 2 above)
Comment (farshad68) [reply src/main/java/eu/enexa/docker/ContainerManagerImpl.java]: done

File: src/main/java/eu/enexa/module/FileBasedModuleManager.java (hunk 3 above)
Comment (farshad68) [reply src/main/java/eu/enexa/module/FileBasedModuleManager.java]: yes it was just for debugging,  deleted

File: src/main/java/eu/enexa/service/EnexaServiceImpl.java (hunk 4 above)
Comment (farshad68) [reply src/main/java/eu/enexa/service/EnexaServiceImpl.java]: done

File: src/main/java/eu/enexa/sparql/SparqlBasedMetadataManager.java (hunk 6 above)
Comment (farshad68) [reply src/main/java/eu/enexa/sparql/SparqlBasedMetadataManager.java]: you right :D

**Review by MichaelRoeder — APPROVED** [review]
LGTM

**Pull request merged**